modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1288,13 +1288,28 @@ def n_ary_operation(cls, left, func, right: list):
         func = cls.preprocess_func(func)
 
         def get_right_block(right_partitions, row_idx, col_idx):
-            blocks = right_partitions[row_idx][col_idx].list_of_blocks
-            # TODO Resolve this assertion as a part of #4691, because the current implementation assumes
-            # that partition contains only 1 block.
-            assert (
-                len(blocks) == 1
-            ), f"Implementation assumes that partition contains only 1 block, but {len(blocks)} recieved."
-            return blocks[0]
+            partition = right_partitions[row_idx][col_idx]
+            blocks = partition.list_of_blocks
+            """
+            NOTE:
```

Comment:
Shouldn't we do the aligning of two modin dfs in `_copartition` method?

Reply:
I wouldn't refer to this step as "aligning." `_copartition` aligns the partitions so that multiple modin dataframes have the same number of partitions and all the partitions at one coordinate have the same indices along certain axes. This method is solving a different problem of getting a single physical block for binary operations. Also, some methods like `concat` will call `_copartition` but don't need the resulting partitions to have a single block each.